plugins/inputs/elasticsearch_query/aggregation_parser.go

```diff
@@ -0,0 +1,117 @@
+package elasticsearch_query
+
+import elastic "gopkg.in/olivere/elastic.v5"
+
+func (e *ElasticsearchQuery) parseAggregationResult(aggregationQueryList *[]aggregationQueryData, searchResult *elastic.SearchResult) error {
+
+	var measurements = make(map[string][]*aggKey)
+
+	for _, aggregationQuery := range *aggregationQueryList {
+		m := measurements[aggregationQuery.measurement]
+		a := aggKey{function: aggregationQuery.function, field: aggregationQuery.field, name: aggregationQuery.name}
+
+		if m == nil {
+			m = []*aggKey{&a}
+		} else {
+			m = append(m, &a)
+		}
+		measurements[aggregationQuery.measurement] = m
+	}
+
+	// recurse over aggregation results
+	for measurementName, aggKey := range measurements {
+		fields := make(map[string]interface{})
+		tags := make(map[string]string)
+
+		err := e.recurseResponse(aggKey, searchResult.Aggregations, measurementName, fields, tags)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (e *ElasticsearchQuery) parseSimpleResult(measurement string, searchResult *elastic.SearchResult) error {
+
+	fields := make(map[string]interface{})
+	tags := make(map[string]string)
+
+	fields["doc_count"] = searchResult.Hits.TotalHits
+
+	e.acc.AddFields(measurement, fields, tags)
+	return nil
+}
+
+func (e *ElasticsearchQuery) recurseResponse(aggKeys []*aggKey, aggs elastic.Aggregations, measurementName string, fields map[string]interface{}, tags map[string]string) error {
+
+	for _, aggKey := range aggKeys {
+		resp := e.getResponseAggregation(aggKey.function, aggKey.name, aggs)
+
+		if resp != nil {
+
+			switch resp := resp.(type) {
+			case *elastic.AggregationValueMetric:
+
+				// we've found a metric aggregation, add to field map
+				fields[aggKey.name] = float64(0)
+
+				if resp.Value != nil {
+					fields[aggKey.name] = *resp.Value
+				}
+
+			case *elastic.AggregationBucketKeyItems:
+				// we've found a terms aggregation, iterate over the buckets and try to retrieve the inner aggregation values
+				for item := range resp.Buckets {
```

Comment:
Don't you get a bucket when iterating over the buckets ([reference](https://github.com/olivere/elastic/blob/be977d75416e9a352a0853165c9d1786e3d43c51/search_aggs.go#L1251))?